packages/block-editor/src/components/block-settings-menu/index.js

```diff
@@ -23,113 +26,89 @@ import BlockUnknownConvertButton from './block-unknown-convert-button';
 import __experimentalBlockSettingsMenuFirstItem from './block-settings-menu-first-item';
 import __experimentalBlockSettingsMenuPluginsExtension from './block-settings-menu-plugins-extension';
 
-export function BlockSettingsMenu( { clientIds, onSelect } ) {
+export function BlockSettingsMenu( { clientIds } ) {
 	const blockClientIds = castArray( clientIds );
 	const count = blockClientIds.length;
 	const firstBlockClientId = blockClientIds[ 0 ];
 
 	return (
 		<BlockActions clientIds={ clientIds }>
 			{ ( { onDuplicate, onRemove, onInsertAfter, onInsertBefore, canDuplicate, isLocked } ) => (
-				<Dropdown
-					contentClassName="editor-block-settings-menu__popover block-editor-block-settings-menu__popover"
-					position="bottom right"
-					renderToggle={ ( { onToggle, isOpen } ) => {
-						const toggleClassname = classnames( 'editor-block-settings-menu__toggle block-editor-block-settings-menu__toggle', {
-							'is-opened': isOpen,
-						} );
-						const label = isOpen ? __( 'Hide options' ) : __( 'More options' );
-
-						return (
-							<Toolbar controls={ [ {
-								icon: 'ellipsis',
-								title: label,
-								onClick: () => {
-									if ( count === 1 ) {
-										onSelect( firstBlockClientId );
```

Comment:
Note: I could not discover what this was trying to do. I think it may have been related to a much earlier design where toolbars could have been shown without the block having already been selected. As far as I can tell, this is no longer the case. And if it were, I would suggest instead to enhance the [existing `onFocus` behavior of `BlockListBlock`](https://github.com/WordPress/gutenberg/blob/6cb0b3268217020a101561d6f9b7d37ab08fa7da/packages/block-editor/src/components/block-list/block.js#L236-L247) instead.